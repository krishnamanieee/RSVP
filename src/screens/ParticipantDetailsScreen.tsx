import React from 'react';
import {View, Text, StyleSheet} from 'react-native';
import {StackNavigationProp} from '@react-navigation/stack';
import Theme from '../theme';
import moment from 'moment';
interface IParticipantDetailsProps {
  data: any;
  navigation: StackNavigationProp<any>;
}

class ParticipantDetailsScreen extends React.PureComponent<
  IParticipantDetailsProps,
  {}
> {
  render(): React.ReactNode {
    const {data} = this.props.route.params;
    return (
      <>
        {this.renderItemView('Name', data.name, 0)}
        {this.renderItemView('Age', data.age, 1)}
        {this.renderItemView('Number of Guest', data.guest, 2)}
        {this.renderItemView('Locality', data.locality, 3)}
        {this.renderItemView(
          'DOB',
          moment(data.DOB).format('dddd, MMMM Do YYYY'),
          4,
        )}
        {this.renderItemView('Address', data.address, 5)}
      </>
    );
  }

  private renderItemView = (title: string, value: string, index: number) => {
    const colorValue: string =
      index % 2 === 0 ? Theme.colors.lightGray : Theme.colors.white;
    const colorTile: string =
      index % 2 === 0 ? Theme.colors.white : Theme.colors.lightGray;
    return (
      <View style={styles.itemContainer}>
        <Text style={[styles.itemTitle, {backgroundColor: colorTile}]}>
          {title}
        </Text>
        <Text style={[styles.itemValue, {backgroundColor: colorValue}]}>
          {value}
        </Text>
      </View>
    );
  };
}

export default ParticipantDetailsScreen;

const styles = StyleSheet.create({
  itemContainer: {
    flexDirection: 'row',
    justifyContent: 'center',
  },
  itemTitle: {
    backgroundColor: Theme.colors.lightGray,
    width: Theme.viewPort.width / 3,
    paddingHorizontal: 5,
    paddingVertical: 10,
  },
  itemValue: {
    flex: 1,
    paddingHorizontal: 5,
    paddingVertical: 10,
  },
});
