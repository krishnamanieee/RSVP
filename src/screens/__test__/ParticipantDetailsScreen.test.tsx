import React from 'react';
import {shallow} from 'enzyme';
import renderer from 'react-test-renderer';
import ParticipantDetailsScreen, {
  IParticipantDetailsProps,
} from '../ParticipantDetailsScreen';

jest.mock('react-native-gesture-handler');

const testProps = (Props: IParticipantDetailsProps) => ({
  ...Props,
});

const mockFunction = jest.fn();

let props: IParticipantDetailsProps = {
  // @ts-ignore
  navigation: mockFunction,
  route: {
    params: {
      data: [],
    },
  },
};

describe('ParticipantDetailsScreen', () => {
  let wrapper: any;
  let instance: ParticipantDetailsScreen;
  beforeEach(() => {
    wrapper = shallow<ParticipantDetailsScreen, any, any>(
      <ParticipantDetailsScreen {...props} />,
    );
    instance = wrapper.instance();
  });

  it('should match match snapshot', () => {
    const tree = renderer
      .create(<ParticipantDetailsScreen {...props} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
