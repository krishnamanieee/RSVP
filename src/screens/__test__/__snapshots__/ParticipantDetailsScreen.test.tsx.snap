// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ParticipantDetailsScreen should match match snapshot 1`] = `
Array [
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "backgroundColor": "#dadada",
            "paddingHorizontal": 5,
            "paddingVertical": 10,
            "width": 250,
          },
          Object {
            "backgroundColor": "#fff",
          },
        ]
      }
    >
      Name
    </Text>
    <Text
      style={
        Array [
          Object {
            "flex": 1,
            "paddingHorizontal": 5,
            "paddingVertical": 10,
          },
          Object {
            "backgroundColor": "#dadada",
          },
        ]
      }
    />
  </View>,
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "backgroundColor": "#dadada",
            "paddingHorizontal": 5,
            "paddingVertical": 10,
            "width": 250,
          },
          Object {
            "backgroundColor": "#dadada",
          },
        ]
      }
    >
      Age
    </Text>
    <Text
      style={
        Array [
          Object {
            "flex": 1,
            "paddingHorizontal": 5,
            "paddingVertical": 10,
          },
          Object {
            "backgroundColor": "#fff",
          },
        ]
      }
    />
  </View>,
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "backgroundColor": "#dadada",
            "paddingHorizontal": 5,
            "paddingVertical": 10,
            "width": 250,
          },
          Object {
            "backgroundColor": "#fff",
          },
        ]
      }
    >
      Number of Guest
    </Text>
    <Text
      style={
        Array [
          Object {
            "flex": 1,
            "paddingHorizontal": 5,
            "paddingVertical": 10,
          },
          Object {
            "backgroundColor": "#dadada",
          },
        ]
      }
    />
  </View>,
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "backgroundColor": "#dadada",
            "paddingHorizontal": 5,
            "paddingVertical": 10,
            "width": 250,
          },
          Object {
            "backgroundColor": "#dadada",
          },
        ]
      }
    >
      Locality
    </Text>
    <Text
      style={
        Array [
          Object {
            "flex": 1,
            "paddingHorizontal": 5,
            "paddingVertical": 10,
          },
          Object {
            "backgroundColor": "#fff",
          },
        ]
      }
    />
  </View>,
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "backgroundColor": "#dadada",
            "paddingHorizontal": 5,
            "paddingVertical": 10,
            "width": 250,
          },
          Object {
            "backgroundColor": "#fff",
          },
        ]
      }
    >
      DOB
    </Text>
    <Text
      style={
        Array [
          Object {
            "flex": 1,
            "paddingHorizontal": 5,
            "paddingVertical": 10,
          },
          Object {
            "backgroundColor": "#dadada",
          },
        ]
      }
    >
      Monday, February 10th 2020
    </Text>
  </View>,
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "backgroundColor": "#dadada",
            "paddingHorizontal": 5,
            "paddingVertical": 10,
            "width": 250,
          },
          Object {
            "backgroundColor": "#dadada",
          },
        ]
      }
    >
      Address
    </Text>
    <Text
      style={
        Array [
          Object {
            "flex": 1,
            "paddingHorizontal": 5,
            "paddingVertical": 10,
          },
          Object {
            "backgroundColor": "#fff",
          },
        ]
      }
    />
  </View>,
]
`;
